src/index.js

```diff
@@ -259,11 +259,38 @@ const Route = ({ component, url, matches }) => {
 	return h(component, { url, matches });
 };
 
+class AsyncRoute extends Component {
+	constructor() {
+		super();
+		this.state = {
+			componentData: null
+		};
+	}
+	componentDidMount(){
+		const componentData = this.props.component();
+		if (componentData instanceof Promise) {
```

Comment:
I usually just check for `componentData && componentData.then` so that any promise lib works without necessarily being global.